src/main/java/org/wso2/carbon/connector/azure/storage/AzureConfig.java

```diff
@@ -0,0 +1,101 @@
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.connector.azure.storage;
+
+import org.apache.synapse.ManagedLifecycle;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.SynapseEnvironment;
+import org.wso2.carbon.connector.azure.storage.connection.AzureStorageConnectionHandler;
+import org.wso2.carbon.connector.azure.storage.connection.ConnectionConfiguration;
+import org.wso2.carbon.connector.azure.storage.exceptions.InvalidConfigurationException;
+import org.wso2.carbon.connector.azure.storage.util.AzureConstants;
+import org.wso2.carbon.connector.azure.storage.util.AzureUtil;
+import org.wso2.carbon.connector.azure.storage.util.Error;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.connector.core.connection.ConnectionHandler;
+import org.wso2.carbon.connector.core.util.ConnectorUtils;
+
+public class AzureConfig extends AbstractConnector implements ManagedLifecycle {
+
+    @Override
+    public void init(SynapseEnvironment synapseEnvironment) {
+        // Do nothing on deployment - configs unknown by that time
+    }
+
+    @Override
+    public void destroy() {
+
+        ConnectionHandler.getConnectionHandler().
+                shutdownConnections(AzureConstants.CONNECTOR_NAME);
+    }
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+
+        String connectorName = AzureConstants.CONNECTOR_NAME;
+        String connectionName = "";
+        try {
+            ConnectionConfiguration configuration = getConnectionConfigFromContext(messageContext);
+            connectionName = configuration.getConnectionName();
+
+            ConnectionHandler handler = ConnectionHandler.getConnectionHandler();
+            if (!handler.checkIfConnectionExists(connectorName, connectionName)) {
+                AzureStorageConnectionHandler azureStorageConnectionHandler = new AzureStorageConnectionHandler(configuration);
+                handler.createConnection(AzureConstants.CONNECTOR_NAME, connectionName, azureStorageConnectionHandler);
+            } else {
+                AzureStorageConnectionHandler connectionHandler = (AzureStorageConnectionHandler) handler
+                        .getConnection(connectorName, connectionName);
+                if (connectionHandler.getConnectionConfig() != configuration) {
```

Comment:
do we need reference equality here or object equality ?

Reply:
Fixed with 8306278